### PR TITLE
Allow to delete custom label position

### PR DESCRIPTION
**Pull request by 3nids**
by pressing DEL key when moving the label

### CODE REVIEW

**Comment by nirvn** [comment]
Nice UX improvement. Would be worth applying the same logic to the rotate label tool.

File: src/app/qgsmaptoolmovelabel.cpp [hunk 1]
```diff
@@ -243,10 +243,49 @@ void QgsMapToolMoveLabel::canvasPressEvent( QgsMapMouseEvent *e )
 
 void QgsMapToolMoveLabel::keyReleaseEvent( QKeyEvent *e )
 {
-  if ( mLabelRubberBand && e->key() == Qt::Key_Escape )
+  if ( mLabelRubberBand )
   {
-    // escape is cancel
-    deleteRubberBands();
+    if ( e->key() == Qt::Key_Delete )
```
Comment (nyalldawson) [inline src/app/qgsmaptoolmovelabel.cpp]: switch (e->key() )

File: src/app/qgsmaptoolmovelabel.cpp [hunk 2]
```diff
@@ -243,10 +243,49 @@ void QgsMapToolMoveLabel::canvasPressEvent( QgsMapMouseEvent *e )
...
+          {
+            // if the edit command fails, it's likely because the label x/y is being stored in a physical field (not a auxiliary one!)
+            // and the layer isn't in edit mode
+            if ( !vlayer->isEditable() )
+            {
+              QgisApp::instance()->messageBar()->pushWarning( tr( "Delete Label Position" ), tr( "Layer “%1” must be editable in order to move labels from it" ).arg( vlayer->name() ) );
+            }
+            else
+            {
+              QgisApp::instance()->messageBar()->pushWarning( tr( "Delete Label Position" ), tr( "Error encountered while storing new label position" ) );
```
Comment (nyalldawson) [inline src/app/qgsmaptoolmovelabel.cpp]:
```suggestion
              QgisApp::instance()->messageBar()->pushWarning( tr( "Delete Label Position" ), tr( "Error encountered while removing stored label position" ) );
```

File: src/app/qgsmaptoolmovelabel.cpp [hunk 3]
```diff
@@ -243,10 +243,49 @@ void QgsMapToolMoveLabel::canvasPressEvent( QgsMapMouseEvent *e )
...
+          vlayer->beginEditCommand( tr( "Delete Label Position" ) + QStringLiteral( " '%1'" ).arg( currentLabelText( 24 ) ) );
+          bool success = vlayer->changeAttributeValue( mCurrentLabel.pos.featureId, xCol, QVariant() );
+          success = vlayer->changeAttributeValue( mCurrentLabel.pos.featureId, yCol, QVariant() ) && success;
+          if ( !success )
+          {
+            // if the edit command fails, it's likely because the label x/y is being stored in a physical field (not a auxiliary one!)
+            // and the layer isn't in edit mode
+            if ( !vlayer->isEditable() )
+            {
+              QgisApp::instance()->messageBar()->pushWarning( tr( "Delete Label Position" ), tr( "Layer “%1” must be editable in order to move labels from it" ).arg( vlayer->name() ) );
```
Comment (nyalldawson) [inline src/app/qgsmaptoolmovelabel.cpp]:
```suggestion
              QgisApp::instance()->messageBar()->pushWarning( tr( "Delete Label Position" ), tr( "Layer “%1” must be editable in order to remove stored label positions" ).arg( vlayer->name() ) );
```

**Review by nyalldawson — APPROVED** [review]
Looks good!

**Comment by nirvn** [comment]
@3nids , thanks for going the extra mile here and implementing the behavior for the rotation tool :+1:

**Pull request merged**